### PR TITLE
Rebuild Nginx with mod_zip on top of nginx:alpine

### PR DESCRIPTION
**Pull request by vbrown608**
This is for Redmine #19718

* Switches the OS to Apline
* Rebuilds Nginx on top of the mainline Nginx Alpine image. That lets us take advantage some of the [setup performed in the base image](https://github.com/nginxinc/docker-nginx/blob/97b65112180e0c7764465aa47a974fc7af3c99ae/mainline/alpine/Dockerfile) including setting `NGINX_VERSION`.

This is currently deployed on staging.

(CI won't pass until #549 lands)

### CODE REVIEW

File: nginx/nginx.conf [hunk 1]
```diff
@@ -1,25 +1,26 @@
+user nginx;
 worker_processes auto;
-pid /run/nginx.pid;
-daemon off;
+
+error_log  /var/log/nginx/error.log warn;
```
Comment (vbrown608) [inline nginx/nginx.conf]: Redirected here https://github.com/nginxinc/docker-nginx/blob/97b65112180e0c7764465aa47a974fc7af3c99ae/mainline/alpine/Dockerfile#L136-L137

**Comment by wioux** [comment]
Awesome! :whale:

File: nginx/Dockerfile [hunk 2]
```diff
@@ -38,11 +36,8 @@ RUN apt-get update \
     && cd .. \
     && rm -f nginx-${NGINX_VERSION}.tar.gz \
     && rm -rf nginx-${NGINX_VERSION} \
+    && apk del .build-deps \
```
Comment (wioux) [inline nginx/Dockerfile]: TIL. What a nice feature.

Comment (vbrown608) [reply nginx/Dockerfile]: :D I thought that was neat too! I saw it in the Dockerfile for the base image.

**Pull request merged**